stdlib/REPL/src/REPL.jl

```diff
@@ -380,7 +417,7 @@ function display(d::REPLDisplay, mime::MIME"text/plain", x)
             # this can override the :limit property set initially
             io = foldl(IOContext, d.repl.options.iocontext, init=io)
         end
-        show(io, mime, x[])
+        show_limited(io, mime, x[])
```

Comment:
Is there any noticable performance impact on this, say to show a big RGB colored unicode image in the terminal? From https://github.com/JuliaImages/ImageInTerminal.jl for example.

Reply:
yikes, it is much slower, and noticeably so, even for non-giant images:

```julia
julia> using Images, TestImages, ImageInTerminal, REPL

julia> @time show(stdout, MIME"text/plain"(), testimage("cameraman"));
...image...
0.002926 seconds (16.24 k allocations: 2.104 MiB)

julia> @time REPL.show_limited(stdout, MIME"text/plain"(), testimage("cameraman"));
...image...
 2.656497 seconds (57.16 k allocations: 3.050 MiB)
```

Reply:
probably printing 1 byte at a time is the issue, but I don't know how to actually optimize it...

Reply:
I added an `unsafe_write` method and now performance looks fine:
```julia
julia> @btime REPL.show_limited(devnull, MIME"text/plain"(), testimage("cameraman"));
  1.172 ms (11849 allocations: 1.97 MiB)

julia> @btime REPL.show(devnull, MIME"text/plain"(), testimage("cameraman"));
  1.170 ms (11844 allocations: 1.97 MiB)
```
or without devnull:
```julia
julia> @time REPL.show_limited(stdout, MIME"text/plain"(), testimage("cameraman"));
...omitted image...
0.002279 seconds (11.75 k allocations: 1.965 MiB)
```